src/crate/client/doctests/cursor.txt

```diff
@@ -296,6 +304,69 @@ closed connection an ``ProgrammingError`` exception will be raised::
     ...
     crate.client.exceptions.ProgrammingError: Cursor closed
 
+
+Python data type conversion
+===========================
+
+The cursor object can optionally convert database types to native Python data
+types. Currently, this is implemented for the CrateDB data types ``IP`` and
+``TIMESTAMP`` on behalf of the ``DefaultTypeConverter``.
+
+::
+
+    >>> cursor = connection.cursor(converter=Cursor.get_default_converter())
+
+    >>> connection.client.set_next_response({
+    ...     "col_types": [4, 5, 11],
+    ...     "rows":[ [ "foo", "10.10.10.1", 1658167836758 ] ],
+    ...     "cols":[ "name", "address", "timestamp" ],
+    ...     "rowcount":1,
+    ...     "duration":123
+    ... })
+
+    >>> cursor.execute('')
+
+    >>> cursor.fetchone()
+    ['foo', IPv4Address('10.10.10.1'), datetime.datetime(2022, 7, 18, 18, 10, 36, 758000)]
+
+
+Custom data type conversion
+===========================
+
+By providing a custom converter instance, you can define your own data type
+conversions. For investigating the list of available data types, please either
+inspect the ``DataType`` enum, or the documentation about the list of available
+`CrateDB data type identifiers for the HTTP interface`_.
+
+To create a simple converter for converging CrateDB's ``BIT`` type to Python's
+``int`` type::
+
+    >>> from crate.client.converter import Converter, DataType
+
+    >>> converter = Converter()
+    >>> converter.set(DataType.BIT, lambda value: int(value[2:-1], 2))
+    >>> cursor = connection.cursor(converter=converter)
+
+Proof that the converter works correctly, ``B\'0110\'`` should be converted to
+``6``. CrateDB's ``BIT`` data type has the numeric identifier ``25``::
+
+    >>> connection.client.set_next_response({
+    ...     "col_types": [25],
+    ...     "rows":[ [ "B'0110'" ] ],
+    ...     "cols":[ "value" ],
+    ...     "rowcount":1,
+    ...     "duration":123
+    ... })
+
+    >>> cursor.execute('')
+
+    >>> cursor.fetchone()
+    [6]
```

Comment:
Do you think the type converter for the `BIT` data type should be included into the `DefaultTypeConverter`, like the converters for `IP` and `TIMESTAMP`? I am talking about this one:

https://github.com/crate/crate-python/blob/33ecbf36a3b50606f5138ce9dcee6b00b81996c0/src/crate/client/doctests/cursor.txt#L347

Reply:
I'd leave it out for now. Easy enough to add later if requested by users